lucille-core/src/test/java/com/kmwllc/lucille/stage/ElasticsearchLookupTest.java

```diff
@@ -0,0 +1,208 @@
+    Function<GetRequest.Builder, ObjectBuilder<GetRequest>> fn = (g) -> g.index(null).id("doc");
+    ObjectNode example = new ObjectNode(JsonNodeFactory.instance);
+    example.put("foo", "bar");
+    example.put("foo2", 10);
+    ObjectNode base = new ObjectNode(JsonNodeFactory.instance);
+    base.put("foo", "empty");
+    base.put("id", "doc");
+    Document doc = Document.createFromJson(base.toString());
+    Document doc2 = Document.createFromJson(base.toString());
+    Document doc3 = Document.createFromJson(base.toString());
+    Document doc4 = Document.createFromJson(base.toString());
+    Document doc5 = Document.createFromJson(base.toString());
+    Document doc6 = Document.createFromJson(base.toString());
+    try (MockedStatic<ElasticsearchUtils> mockedUtils = Mockito.mockStatic(ElasticsearchUtils.class)) {
+
+      GetResponse<ObjectNode> notFound = (GetResponse<ObjectNode>) mock(GetResponse.class);
+      GetResponse<ObjectNode> found = (GetResponse<ObjectNode>) mock(GetResponse.class);
+
+      when(notFound.found()).thenReturn(false);
+      when(found.found()).thenReturn(true);
+      when(found.source()).thenReturn(example);
+
+      ElasticsearchClient throwing = mock(ElasticsearchClient.class);
+      ElasticsearchClient noResponse = mock(ElasticsearchClient.class);
+      ElasticsearchClient foundClient = mock(ElasticsearchClient.class);
+
+      when(throwing.get((Function<GetRequest.Builder, ObjectBuilder<GetRequest>>) Mockito.any(), Mockito.any()))
+          .thenThrow(new IOException("get failed"));
+      when(noResponse.get((Function<GetRequest.Builder, ObjectBuilder<GetRequest>>) Mockito.any(), (Class)Mockito.any()))
+          .thenReturn(notFound);
+      when(foundClient.get((Function<GetRequest.Builder, ObjectBuilder<GetRequest>>) Mockito.any(), (Class)Mockito.any()))
+          .thenReturn(found);
+
+      mockedUtils.when(() -> ElasticsearchUtils.getElasticsearchOfficialClient(config)).thenReturn(throwing);
+      mockedUtils.when(() -> ElasticsearchUtils.getElasticsearchOfficialClient(config2)).thenReturn(noResponse);
+      mockedUtils.when(() -> ElasticsearchUtils.getElasticsearchOfficialClient(config3)).thenReturn(foundClient);
+      mockedUtils.when(() -> ElasticsearchUtils.getElasticsearchOfficialClient(config4)).thenReturn(foundClient);
+      mockedUtils.when(() -> ElasticsearchUtils.getElasticsearchOfficialClient(config5)).thenReturn(foundClient);
+      mockedUtils.when(() -> ElasticsearchUtils.getElasticsearchOfficialClient(config6)).thenReturn(foundClient);
+
+      assertThrows(StageException.class, () -> {
+      new ElasticsearchLookup(config).processDocument(doc);
+      });
+      new ElasticsearchLookup(config2).processDocument(doc2);
+      new ElasticsearchLookup(config3).processDocument(doc3);
+      new ElasticsearchLookup(config4).processDocument(doc4);
+      new ElasticsearchLookup(config5).processDocument(doc5);
+      new ElasticsearchLookup(config6).processDocument(doc6);
```

Comment:
Difficult to read. Six different configs. Six different docs. Naming of configs and docs is numerical and doesn't give any clues about what cases they represent. To understand the setup, you have to go look in multiple different places above. No comments saying what the different cases are.

Could you reorganize this test and the others so that the setup logic for each case we're testing is grouped together and maybe there's a code comment saying what the case is trying to test? I think this could be done without any new code, just by reorganizing the code you've already written here.

Looks like very strong & comprehensive test coverage here... but we should make tests as easy to read as possible.